cloudflare.loader.php

```diff
@@ -43,8 +42,24 @@
     // Load Deactivation Script
     register_deactivation_hook(CLOUDFLARE_PLUGIN_DIR.'cloudflare.php', array($cloudflareHooks, 'deactivate'));
 
-    // Load Automatic Cache Purge
-    add_action('switch_theme', array($cloudflareHooks, 'purgeCache'));
-    add_action('customize_save_after', array($cloudflareHooks, 'purgeCache'));
-    add_action('save_post', array($cloudflareHooks, 'purgePage'));
+    // add_action('admin_bar_menu', array($cloudflareHooks, 'purgeCacheButton'), 100);
+}
+
+// Load Automatic Cache Purge
+// "purgeCache" callback purges everything
+add_action('switch_theme', array($cloudflareHooks, 'purgeCache'));
+add_action('customize_save_after', array($cloudflareHooks, 'purgeCache'));
+
+$cloudflarePurgeActions = array(
+    'autoptimize_action_cachepurged',   // Compat with https://wordpress.org/plugins/autoptimize/
+    'save_post',                        // Save a post
+    'deleted_post',                     // Delete a post
+    'trashed_post',                     // Empty Trashed post
+    // 'edit_post',                        // Edit a post - includes leaving comments
```

Comment:
edit_post should be called when there's a change in the post itself. I mainly used it for purging when someone comments on the post but it has a bug so I'll look at it the future.